File: 4week/2task/Mohican/Mohican.cpp
#include <iostream>
#include "Mohican.h"

Mohican::Mohican() {
    counter += 1;
    name = "";
    lastMohican = this;
}

Mohican::Mohican(Mohican& mohican) {
    counter += 1;
    name = "";
    lastMohican = this;
}

int Mohican::getCounter() { 
    return counter;
}

std::string Mohican::getName() { 
    return this->name;
}

void Mohican::setName(std::string name) { 
    this->name = name;
}

std::string Mohican::getLastMohican() { 
    return lastMohican->name;
}

int Mohican::counter = 0;
Mohican* Mohican::lastMohican = NULL;

File: 4week/3task/Mohican/Mohican.java
public class Mohican {
    private String name;
    private static Mohican lastMohican;
    private static Mohican prevMohican;

    public static Mohican getLast() {
        return lastMohican;
    }

    public Mohican(String name) {
        this.name = name;
        prevMohican = lastMohican;
        lastMohican = this;
    }

     public Mohican(Mohican other) {
        this.name = other.name;
    }

    public String getName() {
        return name;
    }

    public String toString() {
        return "" + name;
    }
}
